modules/auxiliary/scanner/http/nagios_xi_scanner.rb

```diff
@@ -0,0 +1,214 @@
+
+      if error_message
+        print_error("#{rhost}:#{rport} - #{error_message}")
+        print_warning('Please provide a valid Nagios XI USERNAME and PASSWORD, or a specific VERSION to check')
+        return
+      end
+
+      print_status("Target is Nagios XI with version #{nagios_version_result}")
+
+      # check if the Nagios XI version matches any exploit modules
+      return rce_check(nagios_version_result, true)
+
+    end
+
+    # use nagios_xi_login to try and authenticate
+    # if authentication succeeds, nagios_xi_login returns an array containing the http response body of a get request to index.php and the session cookies
+    login_result, error_message = nagios_xi_login(username, password, finish_install)
+    case login_result
+    when 1..3 # an error occurred
+      print_error("#{rhost}:#{rport} - #{error_message}")
+      return
+    when 4 # Nagios XI is not fully installed
+      install_result = install_nagios_xi(password)
+      if install_result
+        print_error("#{rhost}:#{rport} - #{install_result[1]}")
+        return
+      end
+
+      login_result, error_message = login_after_install_or_license(username, password, finish_install)
+      case login_result
+      when 1..3 # an error occurred
+        print_error("#{rhost}:#{rport} - #{error_message}")
+        return
+      when 4 # Nagios XI is still not fully installed
+        print_error("#{rhost}:#{rport} - Failed to install Nagios XI on the target.")
+        return
+      end
+    end
+
+    # when 5 is excluded from the case statement above to prevent having to use this code block twice
+    # including when 5 would require using this code block once at the end of the `when 4` code block above, and once here
+    if login_result == 5 # the Nagios XI license agreement has not been signed
+      auth_cookies, nsp = error_message
+      sign_license_result = sign_license_agreement(auth_cookies, nsp)
+      if sign_license_result
+        print_error("#{rhost}:#{rport} - #{sign_license_result[1]}")
+        return
+      end
+
+      login_result, error_message = login_after_install_or_license(username, password, finish_install)
+      case login_result
+      when 1..3
+        print_error("#{rhost}:#{rport} - #{error_message}")
+        return
+      when 5 # the Nagios XI license agreement still has not been signed
+        print_error("#{rhost}:#{rport} - Failed to sign the license agreement.")
+        return
+      end
+    end
+
+    print_good('Successfully authenticated to Nagios XI')
+
+    # obtain the Nagios XI version
+    nagios_version = nagios_xi_version(login_result)
+    if nagios_version.nil?
+      print_error("#{rhost}:#{rport} - Unable to obtain the Nagios XI version from the dashboard")
+      return
+    end
+
+    print_status("Target is Nagios XI with version #{nagios_version}")
+
+    # check if the Nagios XI version matches any exploit modules
+    return rce_check(nagios_version, true)
```

Comment:
Last statement of the function and this function doesn't really return anything. Suggest writing it this way:

```suggestion
    rce_check(nagios_version, true)
```